programs/drift/src/controller/orders.rs

```diff
@@ -3764,6 +3773,374 @@ pub fn fulfill_spot_order_with_match(
     Ok(base_asset_amount)
 }
 
+pub fn fulfill_spot_order_with_phoenix(
+    base_market: &mut SpotMarket,
+    quote_market: &mut SpotMarket,
+    taker: &mut User,
+    taker_stats: &mut UserStats,
+    taker_order_index: usize,
+    taker_key: &Pubkey,
+    filler: Option<&mut User>,
+    filler_stats: Option<&mut UserStats>,
+    filler_key: &Pubkey,
+    now: i64,
+    slot: u64,
+    oracle_map: &mut OracleMap,
+    _fee_structure: &FeeStructure,
+    phoenix_fulfillment_params: &mut PhoenixFulfillmentParams,
+) -> DriftResult<u64> {
+    // Begin copy-paste
```

Comment:
These comments will be removed, but use them to skip parts of the code that are duplicated from the Serum order handling